src/app/core/guards/auth.guard.ts

```diff
@@ -0,0 +1,22 @@
+import { Injectable } from '@angular/core';
+import { CanActivate, ActivatedRouteSnapshot, RouterStateSnapshot, UrlTree, Router } from '@angular/router';
+import { Observable } from 'rxjs';
+
+import { AuthenticationService } from '../service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthGuard implements CanActivate {
+  constructor(
+    private authService: AuthenticationService,
+    private router: Router
+  ) {}
+  canActivate(
+    next: ActivatedRouteSnapshot,
+    state: RouterStateSnapshot): Observable<boolean | UrlTree> | Promise<boolean | UrlTree> | boolean | UrlTree {
+      const authPageUrlTree = this.router.parseUrl("/login");
+
+      return this.authService.isAuthenticated() ? true : authPageUrlTree;
```

Comment:
just return this.authService.isAuthenticated()

you don't need to check if it's login page. The guard should be URL agnostic.  
Maybe that's why I get logged out when refreshing the page?)

Reply:
@SteveRusin No, redirecting to login page on refreshing was for the reason of not correct working of isAuthenticated() service method.

But in this case how I'll make redirection to login page?